workflow/controller/operator.go

```diff
@@ -198,6 +198,11 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 
 	woc.log.Infof("Processing workflow")
 
+	if woc.wf.Status.Fulfilled() {
```

Comment:
Yes, this is a better fix than I suggested, it addresses the issue where labels and status are inconsistent, presumably that is what you think the underlying cause is. 

Would it be sensible to patch the labels at this point?